app/streams/publishing_api/item_handler.rb

```diff
@@ -0,0 +1,53 @@
+class PublishingAPI::ItemHandler
+  def initialize(old_item:, new_item:, subpage:)
+    @old_item = old_item
+    @new_item = new_item
+    @subpage = subpage
+
+    # old_item may be nil if the item belongs to a brand new document
+    # new_item may be nil if a base_path is reused for a different content id
```

Comment:
It is not clear to me how the `base_path` can be nil. If reusing a `base_path` implies that we need to guard agains a `nil` issue at this point, I would try to detect earlier, and make it explicit in our code.